### ISSUE TITLE
Invalid parameters (invalid amount)

### ISSUE THREAD

**Issue by hengboyz** [tjwoon/csIPay88#7]
Hi, 
I am sorry, as I am new to this but I have no idea why it always show me this error message invalid parameters (invalid amount) when i tried to submit the payment. I have tried all possible value i can think of. (eg. 100, 1, 1.00 as string, int, or double)...

**Comment by tjwoon** [comment]
Where do you get the message?

Post your function call.

**Comment by hengboyz** [comment]
Hi,

i call the function as told, please see below:

makePayment(): void {
    cloudSky.iPay88.makePayment(
    {
        amount: "100",  // Trying to charge 1.00
        name: "alan",
        email: "heng",
        phone: "+60123456789",
        refNo: "A120",
        currency: "MYR",
        lang: "UTF-8",
        country: "MY",
        description: "test sales",
        remark: "",
        paymentId: "2",
        merchantKey: "",
        merchantCode: "",
        backendPostUrl: "" // The URL which iPay will call from their 
            // servers upon successful payment. 
    },
    function (resp) {
      alert(resp.authCode);
        // Success callback 
        // resp = { 
        //     transactionId: transId, 
        //     referenceNo: refNo, 
        //     amount: amount, 
        //     remarks: remarks, 
        //     authCode: auth, 
        // } 
    },
    function (err) {
      alert(err.amount);
      alert(err.err);
        // Failure callback 
        // err = "some error string" OR 
        // err = { 
        //     transactionId: transId, 
        //     referenceNo: refNo, 
        //     amount: amount, 
        //     remarks: remarks, 
        //     err: error message, // "canceled" if user canceled the payment. 
        // } 
    })
  }

after i click the PAY button and call the above function, this is the get message i get....

![screenshot_1510630859](https://user-images.githubusercontent.com/1905064/32761902-0b6bec90-c931-11e7-97f6-5d372f801426.png)

**Comment by tjwoon** [comment]
Amount should be a number, not a string. Refer to https://github.com/tjwoon/csIPay88#usage

Please try:

```
makePayment(): void {
    cloudSky.iPay88.makePayment(
        {
            amount: 100, // Trying to charge 1.00
            name: "alan",
            ...
```

Btw are you using this with Typescript? Did you create some typings files for it? If you did could you share them?

TJ

**Comment by hengboyz** [comment]
Yea, i tried number too, but same error i got.

Yes, this is TypeScript... This actually my first time using Ionic...

Below is the one in my ts file:
```
import { Component } from '@angular/core';
import { IonicPage, NavController, NavParams } from 'ionic-angular';
import { GLOBALS } from '../../Util/utility';
import { UserProvider } from "../../providers/user/user";
import { LoadingController } from "ionic-angular";

declare var cloudSky: any;

@IonicPage()
@Component({
  selector: 'page-ipay88',
  templateUrl: 'ipay88.html',
})
export class Ipay88Page {

  order_total = 0;
  order_id = 0;
  payment_method = 0;
  globals : any = GLOBALS;

  currentUserObjID = this.getCurrentUserId();
 
  constructor(public navCtrl: NavController, public navParams: NavParams, private userProvider: UserProvider, 
              public loadingCtrl: LoadingController) {
    this.order_total = this.navParams.get('order_amt');
    this.order_id = this.navParams.get('order_id');
  }

  ionViewDidLoad() {
    console.log('ionViewDidLoad Ipay88Page');
  }

  getCurrentUserId(): string {
    let user = [];
    user = this.userProvider.currentUser();

    if(user != null) {
      console.dir(user['id']);
      return user['id'];
    }
    return null;
  }

  makePayment(): void {
    cloudSky.iPay88.makePayment(
    {
        amount: 100,
        name: "alan",
        email: "heng@gmail.com",
        phone: "+60123456789",
        refNo: "A120",
        currency: "MYR",
        lang: "UTF-8",
        country: "MY",
        description: "test sales",
        remark: "",
        paymentId: "2",
        merchantKey: "XXXXX",
        merchantCode: "XXXXX",
        backendPostUrl: "XXXXXX" // The URL which iPay will call from their 
            // servers upon successful payment. 
    },
    function (resp) {
      alert(resp.authCode);
        // Success callback 
        // resp = { 
        //     transactionId: transId, 
        //     referenceNo: refNo, 
        //     amount: amount, 
        //     remarks: remarks, 
        //     authCode: auth, 
        // } 
    },
    function (err) {
      alert(err.amount);     //<<<<<< **This line alert me it received the amount as 0**
      alert(err.err);             // <<<<< **This is where the error message it gives**
        // Failure callback 
        // err = "some error string" OR 
        // err = { 
        //     transactionId: transId, 
        //     referenceNo: refNo, 
        //     amount: amount, 
        //     remarks: remarks, 
        //     err: error message, // "canceled" if user canceled the payment. 
        // } 
    })
  }
}
```

In my html template file, I just added one button to make the payment and call the makePayment() function

`<button padding ion-button full (click)="makePayment()">Pay Now</button>`

**Comment by ManigandanRaamanathan** [comment]
Did you fixed this issue?

**Comment by hengboyz** [comment]
No... i didnt... the reply is too slow and even no reply after a long time... so i gave it up. you face the same issue?

**Comment by tjwoon** [comment]
Hi guys. Sorry for the late response.

@hengboyz your function call is similar to mine (same types, etc). I am not facing the problem you have.

Can you or @ManigandanRaamanathan request iPay88 to check what "amount" they received? They should have logs of each attempted transaction.

Based on [this code](https://github.com/tjwoon/csIPay88/blob/master/android/IPay88.java#L237), I am formatting the "amount" integer to a string with 2 decimal places before sending to iPay88. I don't see any problem with the code, but if I can see what iPay88 received in your case, I might find a bug which can be fixed.
